torch/csrc/Size.cpp

```diff
@@ -23,10 +23,19 @@ static PyObject * THPSize_pynew(PyTypeObject *type, PyObject *args, PyObject *kw
   if (self) {
     for (Py_ssize_t i = 0; i < PyTuple_Size(self); ++i) {
       PyObject *item = PyTuple_GET_ITEM(self.get(), i);
-      if (!THPUtils_checkLong(item)) {
-        return PyErr_Format(PyExc_TypeError, "torch.Size() takes an iterable of 'int' (item %zd is '%s')",
-            i, Py_TYPE(item)->tp_name);
+      if (THPUtils_checkLong(item)) {
+        continue;
       }
+      // Try replacing item with the result of item.__index__()
+      THPObjectPtr number(PyNumber_Index(item));
```

Comment:
is this ref counting correct?  PyNumber_Index documentation doesn't indicate that it returns a new reference, but the cpython source code seems to indicate it does...

Reply:
My understanding is that it returns a new reference. I also used the cpython source code as reference: https://github.com/python/cpython/blob/f320be77ffb73e3b9e7fc98c37b8df3975d84b40/Objects/abstract.c#L1251-L1289